File: decode.js
const base64 = "eyJleHBpcmVzX2F0IjogIjIwMjYtMDktMDJUMTQ6MDU6NDYuMzI3MjkxIiwgIm1hY2hpbmVfaWQiOiAiNTM0NDcxNjE2NDk5MzcifW6A5Zo3vG-dkE1dvO4jkCp5epbOvZ855ve4yr93f3p4";
// const base64 = "dkE1dvO4jkCp5epbOvZ855ve4yr93f3p4";



// Split payload (before the signature)
const [payloadB64] = base64.split(/[\-_.]/); // crude split, depends on format
const decoded = Buffer.from(payloadB64, "base64").toString("utf8");

console.log(decoded)

const data = JSON.parse(decoded);
console.log("Decoded:", data);

const now = new Date();
const expiry = new Date(data.expires_at);

if (now > expiry) {
  console.log("❌ Token expired");
} else {
  console.log("✅ Token valid until", expiry.toISOString());
}

File: decode.py
import base64
key = "eyJleHBpcmVzX2F0IjogIjIwMjYtMTAtMDNUMDQ6NTU6NTguMjAzMDU0IiwgIm1hY2hpbmVfaWQiOiAiOTYyNjUwODgyMTIyIn1-uH8oc1fKnlcRVTNzhShQimSQSQCLsjj_5Fv07mupaA"
padded = key + "=" * ((4 - len(key) % 4) % 4)
data = base64.urlsafe_b64decode(padded.encode())
print(data[:-32].decode())   # payload
print(len(data[-32:]))       # signature length
